sdk/helper/template/funcs.go

```diff
@@ -0,0 +1,68 @@
+package template
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	UUID "github.com/hashicorp/go-uuid"
+)
+
+func nowSeconds() string {
+	return strconv.FormatInt(time.Now().Unix(), 10)
+}
+
+func nowNano() string {
+	return strconv.FormatInt(time.Now().UnixNano(), 10)
+}
+
+func truncate(maxLen int, str string) (string, error) {
+	if maxLen <= 0 {
+		return "", fmt.Errorf("max length must be > 0 but was %d", maxLen)
+	}
+	if len(str) > maxLen {
+		return str[:maxLen], nil
+	}
+	return str, nil
+}
+
+const (
+	sha256HashLen = 8
+)
+
+func truncateSHA256(maxLen int, str string) (string, error) {
+	if maxLen <= 8 {
+		return "", fmt.Errorf("max length must be > 8 but was %d", maxLen)
+	}
+
+	if len(str) <= maxLen {
+		return str, nil
+	}
+
+	truncIndex := maxLen - sha256HashLen
+	hash := hashSHA256(str[truncIndex:])
+	result := fmt.Sprintf("%s%s", str[:truncIndex], hash[:sha256HashLen])
+	return result, nil
+}
+
+func hashSHA256(str string) string {
+	return fmt.Sprintf("%x", sha256.Sum256([]byte(str)))
+}
+
+func uppercase(str string) string {
+	return strings.ToUpper(str)
+}
+
+func lowercase(str string) string {
+	return strings.ToLower(str)
+}
+
+func replace(find string, replace string, str string) string {
```

Comment:
Nit: Can we have the signature ordering be the same as `strings.Replace`? That is, `str` (the input) becomes the first argument for this func. 

On a similar note, I'm curious why the other funcs like `truncate` take the string input as its last argument.

Reply:
Unfortunately no. The way that the template library passes arguments using `|` seems to require the value being passed to it to be at the end of the argument list.

Reply:
Ah, I recall this from past experience now that you mention it.